### PR TITLE
[WIP] Introduce CheckServiceDefinitionsCommand

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Introduce CheckServiceDefinitionsCommand that attempts to build every available Symfony service for PrestaShop. The goal is to check each of them is buildable. Maybe also add some checks (for example the service definition is valid, to detect [this kind of errors](https://github.com/PrestaShop/PrestaShop/pull/12125) ). It should be called by Travis in the end.
| Type?         | new feature
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | No QA needed as this is a new code safety check

**This PR is in work-in-progress: it will be finished when the Command outputs no issue** (so I have to fix all found issues before the PR can be merged).

To run the Command: `bin/console prestashop:debug:services-container`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12127)

### CODE REVIEW

**Comment by matks** [comment]
Too many Symfony services rely on legacy settings such as current employee, context, shop, currency ... and these settings are only set for WEB context, not CLI. So this command outputs way too much "false positive". I keep the service definitions fixes in https://github.com/PrestaShop/PrestaShop/pull/12139 and close this PR.

**Comment by matks** [comment]
Idea: make it a Debug / Dev controller (like the Graph Visualizer) in order to be able to check the container in WEB context

**Comment by matks** [comment]
As suggested by @mickaelandrieu : https://github.com/matthiasnoback/symfony-service-definition-validator could be related and useful for this topic

File: src/PrestaShopBundle/Command/CheckServiceDefinitionsCommand.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    /**
+     * Load the Symfony container with PrestaShop services
+     *
+     * @return ContainerBuilder
+     *
+     * @todo: be able to target PrestaShop services defined outside of PrestaShopBundle
+     */
+    protected function getPrestaShopContainerBuilder()
+    {
+        $container = new ContainerBuilder();
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Command/CheckServiceDefinitionsCommand.php:90]: I'll take the real service container

**Pull request closed without merging**